warehouse/internal/repo/upload.go

```diff
@@ -1381,3 +1381,69 @@ func (u *Uploads) GetFirstAbortedUploadInContinuousAbortsByDestination(ctx conte
 
 	return abortedUploadsInfo, nil
 }
+
+func (u *Uploads) GetSyncLatencies(ctx context.Context, request model.SyncLatencyRequest) ([]model.LatencyTimeSeriesDataPoint, error) {
+	aggregationSQL := getLatencyAggregationSQL(request.AggregationType)
+
+	var filterSQL strings.Builder
+	filterSQL.WriteString(`destination_id = $1 AND workspace_id = $2 AND status = $3 AND created_at >= $4`)
+	filterParams := []any{request.DestinationID, request.WorkspaceID, model.ExportedData, request.StartTime}
+
+	if request.SourceID != "" {
+		filterSQL.WriteString(` AND source_id = $` + strconv.Itoa(len(filterParams)+1))
+		filterParams = append(filterParams, request.SourceID)
+	}
+
+	intervalParamIndex := len(filterParams) + 1
+	filterParams = append(filterParams, request.AggregationMinutes)
```

Comment:
Value of `len(filterParams)` will be same as `intervalParamIndex` post this append, rt ?

Reply:
Yes.

Reply:
So can we replace `intervalParamIndex` with `len(filterParams)` ?